Makefile

```diff
@@ -107,15 +116,38 @@ clientset: ## Generate a typed clientset
 clean: ## Remove all generated files
 	rm -f bazel-*
 
+.PHONY: all-docker-build
+all-docker-build: $(addprefix sub-docker-build-,$(ALL_ARCH))
+	@echo "updating kustomize image patch file for manager resource"
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"$(CONTROLLER_IMG):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+sub-docker-build-%:
+	$(MAKE) ARCH=$* docker-build
+
 .PHONY: docker-build
 docker-build: generate fmt vet manifests ## Build the docker image for controller-manager
-	docker build . -t ${CONTROLLER_IMG}
+	docker build --build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
 	@echo "updating kustomize image patch file for manager resource"
-	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}"'@' ./config/default/manager_image_patch.yaml
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}-$(ARCH):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+.PHONY:all-push ## Push all the architecture docker images and fat manifest docker image
+all-push: all-docker-push push-manifest
+
+.PHONY:all-docker-push ## Push all the architecture docker images
+all-docker-push: $(addprefix sub-docker-push-,$(ALL_ARCH))
+
+sub-docker-push-%:
+	$(MAKE) ARCH=$* docker-push
+
+.PHONY: push-manifest
+push-manifest: ## Push the fat manifest docker image
+	docker manifest create --amend $(CONTROLLER_IMG):$(TAG) $(shell echo $(ALL_ARCH) | sed -e "s~[^ ]*~$(CONTROLLER_IMG)\-&:$(TAG)~g")
+	@for arch in $(ALL_ARCH); do docker manifest annotate --arch $${arch} ${CONTROLLER_IMG}:${TAG} ${CONTROLLER_IMG}-$${arch}:${TAG}; done
+	docker manifest push --purge ${CONTROLLER_IMG}:${TAG}
 
 .PHONY: docker-push
 docker-push: docker-build ## Push the docker image
-	docker push "$(CONTROLLER_IMG)"
+	docker push $(CONTROLLER_IMG)-$(ARCH):$(TAG)
```

Comment:
Is this a breaking change for providers? If so, we should make sure this is covered in the next release and maybe in the docs.

Reply:
This should not be a breaking change, because the providers will reference the manifest list image and will pull down the linked image for the proper architecture.

Reply:
well, as we are pushing both `$(CONTROLLER_IMG)-$(ARCH):$(TAG)` and fat manifest image `$(CONTROLLER_IMG)-$(TAG)`. In all the manifest files we will refer only `$(CONTROLLER_IMG)-$(TAG)`, so we don't need to worry about any doc changes.